src/preact/vote-buttons.js

```diff
@@ -21,15 +21,8 @@ import {h, Component} from 'preact';
 
 export default class VoteButtons extends Component {
 
-  constructor (props) {
-    super(props);
-
-    this.state = {
-      score: this.props.score
-    };
-
-    this.onVoteUp = this.onVoteUp.bind(this);
-    this.onVoteDown = this.onVoteDown.bind(this);
+  componentWillMount () {
+    this.state.score = this.props.score;
   }
 
```

Comment:
I'd probably keep this in the constructor so that you're not mutating the shape of `this.state`.  I'm not sure about support since this isn't transpiled, but `this.props` is available once `super()` has been invoked, so you can (maybe only in babel land) do this:

```js
class Foo {
  state = {
    score: this.props.score
  };
}
```